cmd/version/cmd.go

```diff
@@ -17,77 +17,104 @@ limitations under the License.
 package version
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 
 	verify "github.com/openshift/rosa/cmd/verify/rosa"
 	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/reporter"
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var (
-	args struct {
-		clientOnly bool
-		verbose    bool
+var args struct {
+	clientOnly bool
+	verbose    bool
+}
+
+type RosaVersionOptions struct {
+	reporter   *reporter.Object
+	verifyRosa verify.VerifyRosa
+}
+
+func NewRosaVersionOptions() (*RosaVersionOptions, error) {
+	verifyRosa, err := verify.NewVerifyRosaOptions()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build rosa verify options : %v", err)
 	}
 
-	Cmd             = makeCmd()
-	delegateCommand = verify.Cmd.Run // used in testing
-)
+	rpt, err := reporter.CreateReporter()
+	if err != nil {
+		return nil, fmt.Errorf("these was a problem creating the reporter: %v", err)
+	}
+
+	return &RosaVersionOptions{
+		verifyRosa: verifyRosa,
+		reporter:   rpt,
```

Comment:
@ciaranRoche This _feels_ like a duplication of what we have on `Runtime` in terms of the reporter and moves us away a little bit from the centralized instantiation of things. Is the alternative for us to pass the `Runtime` around as method parameters?